MAINTAINERS.md

```diff
@@ -0,0 +1,51 @@
+# Helm Maintainers
+
+This document explains the leadership structure of the Kubernetes Helm project, and list the current project maintainers.
```

Comment:
Rather than list the maintainers here, why not point people to the Github team that already lists them? Currently, that team is https://github.com/orgs/kubernetes/teams/deployment-manager-maintainers.